tests/modules/annotations/resources/test_annotation_keywords.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=missing-docstring
+
+from tests import utils
+from tests.modules.annotations.resources import utils as annot_utils
+from tests.modules.asset_groups.resources import utils as sub_utils
+from tests.modules.keywords.resources import utils as keyword_utils
+
+
+def test_keywords_on_annotation(
+    flask_app_client, admin_user, researcher_1, test_clone_asset_group_data, db
+):
+    # pylint: disable=invalid-name
+    from app.modules.annotations.models import Annotation
+    from app.modules.keywords.models import Keyword, KeywordSource
+
+    # this gives us an "existing" keyword to work with
+    keyword_value1 = 'TEST_VALUE_1'
+    keyword = Keyword(value=keyword_value1)
+    with db.session.begin():
+        db.session.add(keyword)
+    assert keyword is not None
+
+    sub_utils.clone_asset_group(
+        flask_app_client,
+        admin_user,
+        researcher_1,
+        test_clone_asset_group_data['asset_group_uuid'],
+    )
+
+    response = annot_utils.create_annotation_simple(
+        flask_app_client,
+        researcher_1,
+        test_clone_asset_group_data['asset_uuids'][0],
+    )
+
+    annotation_guid = response.json['guid']
+    annotation = Annotation.query.get(annotation_guid)
+    assert annotation is not None
+
+    # patch to add *existing* keyword
+    res = annot_utils.patch_annotation(
+        flask_app_client,
+        annotation.guid,
+        researcher_1,
+        [utils.patch_add_op('keywords', str(keyword.guid))],
+    )
+    kw = annotation.get_keywords()
+    assert len(kw) == 1
+    assert kw[0].value == keyword_value1
+
+    # patch to add *new* keyword (by value)
+    keyword_value2 = 'TEST_VALUE_2'
+    res = annot_utils.patch_annotation(
+        flask_app_client,
+        annotation.guid,
+        researcher_1,
+        [
+            utils.patch_add_op(
+                'keywords', {'value': keyword_value2, 'source': KeywordSource.user}
+            )
+        ],
+    )
+    kw = annotation.get_keywords()
+    assert len(kw) == 2
+    # since keyword order is arbitrary, we dont know which this is, but should be one of them!
+    assert kw[0].value == keyword_value2 or kw[1].value == keyword_value2
```

Comment:
Huh, wonder if we can set a default order. It's not a huge deal but it would be nice to enforce alphabetical.

Reply:
i posted an "aside" on this in #devhouston ... it seems like you _can_ ... but yeah we could always sort specifically on the `annotation.get_keywords()` method maybe?

Reply:
Yeah, that sounds good. Seems like the sort of list operation that python would have a super easy path for.